File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/repository/custom/CustomGenericRepository.java
package com.coupang.springframework.data.requery.repository.custom;

import com.coupang.springframework.data.requery.repository.RequeryRepository;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.NoRepositoryBean;

/**
 * Extension of {@link CrudRepository} to be added on a custom repository base class. This tests the facility to
 * implement custom base class functionality for all repository instances derived from this interface and implementation
 * base class.
 *
 * @author debop
 * @since 18. 6. 9
 */
@NoRepositoryBean
public interface CustomGenericRepository<T, ID> extends RequeryRepository<T, ID> {

    /**
     * Custom sample queryMethod.
     */
    T customMethod(ID id);
}

File: spring-data-requery/src/test/kotlin/org/springframework/data/requery/repository/custom/CustomGenericRepository.kt
package org.springframework.data.requery.repository.custom

import org.springframework.data.requery.repository.RequeryRepository

/**
 * org.springframework.data.requery.repository.custom.CustomGenericRepository
 *
 * @author debop
 */
interface CustomGenericRepository<E: Any, ID: Any>: RequeryRepository<E, ID> {

    fun customMethod(id: ID): E
}